src/components/RoomNameInput.js

```diff
@@ -72,45 +59,17 @@ class RoomNameInput extends Component {
         super(props);
         this.state = {
             roomName: props.initialValue,
-            error: '',
         };
 
         this.originalRoomName = props.initialValue;
     }
 
-    componentDidUpdate(prevProps, prevState) {
+    componentDidUpdate(_prevProps, prevState) {
         // As we are modifying the text input, we'll bubble up any changes/errors so the parent component can see it
-        if (prevState.roomName !== this.state.roomName) {
-            this.props.onChangeText(this.state.roomName);
-        }
-        if (prevState.error !== this.state.error) {
-            this.props.onChangeError(this.state.error);
+        if (prevState.roomName === this.state.roomName) {
+            return;
         }
-    }
-
-    /**
-     * @param {String} modifiedRoomName
-     */
-    validateRoomName(modifiedRoomName) {
-        const isExistingRoomName = _.some(
-            _.values(this.props.reports),
-            report => report && report.policyID === this.props.policyID && report.reportName === modifiedRoomName,
-        );
-
-        let error = '';
-
-        // We error if the room name already exists. We don't care if it matches the original name provided in this
-        // component because then we are not changing the room's name.
-        if (isExistingRoomName && modifiedRoomName !== this.originalRoomName) {
-            error = this.props.translate('newRoomPage.roomAlreadyExistsError');
-        }
-
-        // Certain names are reserved for default rooms and should not be used for policy rooms.
-        if (_.contains(CONST.REPORT.RESERVED_ROOM_NAMES, modifiedRoomName)) {
-            error = this.props.translate('newRoomPage.roomNameReservedError');
-        }
-
-        this.setState({error});
+        this.props.onChangeText(this.state.roomName);
     }
```

Comment:
We don't need this block now. Call the `this.props.onChangeText(this.state.roomName);` directly when value changes.

Reply:
We need the state for room name modification.